src/vsphere_cpi/lib/cloud/vsphere/drs_rules/vm_group.rb

```diff
@@ -59,8 +59,37 @@ def delete_vm_groups(vm_group_names)
           group_spec.remove_key = vm_group.name
           group_spec
         end
+
+        # If these vm_groups are getting deleted. Also delete all the
+        # vm-host group rules associated with it.
+        #
+        # Now, there is a potential problem here if anyone creates a rule
+        # outside of CPI using this vm_group, it will be deleted too. To narrow
+        # the scope of this potential bug, CPI will only delete the
+        # vm-host-group-affinity rule type. This is because, currently it only
+        # creates vm-host-affinity rule using vm groups.
+        #
+        # Long Term Solution is to disallow admin to specify their own rule
+        # names and CPI generates a rule name from vm_group and hence will be
+        # able to identify correct rule to be deleted.
+        rules_spec = empty_vm_groups.reduce([]) do |rules_spec, vm_group|
```

Comment:
In general `reduce` in Ruby is discouraged. I'm not exactly sure but I think in this case we can use `each_with_object` instead.